File: backend/internal/domain/dto/request/contact_full_data.go
package dto_request

type AddContactFullDataRequest struct {
	GraphId string         `json:"graphId"`
	Node    AddNodeRequest `json:"node"`
	Edge    AddEdgeRequest `json:"edge"`
}

type ChangeContactFullDataRequest struct {
	GraphId string            `json:"graphId"`
	Node    ChangeNodeRequest `json:"node"`
	Edge    ChangeEdgeRequest `json:"edge"`
}

type DeleteContactFullDataRequest struct {
	GraphId string `json:"graphId"`
	NodeId  string `json:"nodeId"`
	EdgeId  string `json:"edgeId"`
}

File: web-app/src/types/contactFullData.ts
import { AddContactEdge, Edge } from "./edge"
import { AddContactNode, NodeChange } from "./node"
import { BaseResponse } from "./response"

export interface AddContactNodeResponse {
    node: {
        idNode: string
    }
    edge: {
        idEdge: string
    }
}

export interface AddContactFullDataRequest {
    graphId: string
    node: AddContactNode
    edge: AddContactEdge
}
export interface ChangeContactFullDataRequest {
    graphId: string
    node: NodeChange
    edge: Edge
}

export interface DeleteContactFullDataRequest {
    graphId: string
    nodeId: string
    edgeId: string
}

export interface DeleteContactFullDataResponse extends BaseResponse {
}



export interface ChangeContactFullDataResponse extends BaseResponse {

}